src/pages/Compare/index.spec.tsx

```diff
@@ -886,6 +916,191 @@ describe(__filename, () => {
+        buildTree,
+        headVersionId,
+        loadDiff,
+        loadVersionFile,
+        version,
+        ...params,
+      });
+      return {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      };
+    };
+
+    it('dispatches fetchDiff and fetchVersionFile for the next file with diff', () => {
+      const addonId = 10;
+      const baseVersionId = 44;
+      const headVersionId = 55;
+      const currentFile = 'current.json';
+      const nextModifiedFile = 'modified.json';
+      const {
+        _fetchDiff,
+        _fetchVersionFile,
+        dispatchSpy,
+        fakeThunk,
+      } = setUpForPreloadAndRender({
+        addonId,
+        baseVersionId,
+        currentFile,
+        headVersionId,
+        nextModifiedFile,
+      });
+
+      expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchDiff).toHaveBeenCalledWith({
+        addonId,
+        baseVersionId,
+        headVersionId,
+        path: nextModifiedFile,
+      });
+      expect(_fetchVersionFile).toHaveBeenCalledWith({
+        addonId,
+        versionId: headVersionId,
+        path: nextModifiedFile,
+      });
+    });
+
+    it.each([['buildTree'], ['loadVersionFile'], ['loadDiff']])(
+      'does not dispatch fetchDiff or fetchVersionFile for the next file before %s is dispatched',
+      (action) => {
+        const currentFile = 'current.json';
+        const nextModifiedFile = 'next.json';
+        const { _fetchDiff, _fetchVersionFile } = setUpForPreloadAndRender({
+          currentFile,
+          nextModifiedFile,
+          [action]: false,
+        });
+
+        expect(_fetchDiff).not.toHaveBeenCalledWith(
+          expect.objectContaining({
+            path: nextModifiedFile,
+          }),
+        );
+        expect(_fetchVersionFile).not.toHaveBeenCalledWith(
+          expect.objectContaining({
+            path: nextModifiedFile,
+          }),
+        );
+      },
+    );
+
+    it('does not dispatch fetchVersionFile for the next file if it is loading', () => {
+      const headVersionId = 2222;
+      const baseVersionId = 11;
+      const currentFile = 'current.json';
+      const nextModifiedFile = 'next.json';
+      const store = configureStore();
+      store.dispatch(
+        versionsActions.beginFetchVersionFile({
+          path: nextModifiedFile,
+          versionId: headVersionId,
+        }),
+      );
+      const { _fetchVersionFile } = setUpForPreloadAndRender({
+        baseVersionId,
+        currentFile,
+        headVersionId,
+        nextModifiedFile,
+        store,
+      });
+
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith(
+        expect.objectContaining({
+          versionId: headVersionId,
+          path: nextModifiedFile,
+        }),
+      );
+    });
+
+    it('does not dispatch fetchVersionFile for the next file when it is already loaded', () => {
+      const headVersionId = 2222;
+      const baseVersionId = 11;
```

Comment:
It looks like `baseVersionId` isn't used for any setup or assertions.

Reply:
https://github.com/mozilla/addons-code-manager/pull/1085#discussion_r336133628